spec/attribute_spec.cr

```diff
@@ -36,4 +44,38 @@ describe "Avram::Attribute" do
       "Error: attr_name with value of 'fake' is invalid",
     ])
   end
+
+  describe "Attribute#changed?" do
+    it "tests if the value has changed since initialization" do
+      attribute = Avram::Attribute.new(name: :state, param: nil, value: "caterpillar", param_key: "test_form")
+
+      attribute.changed?.should be_false
+      attribute.value = "caterpillar"
+      attribute.changed?.should be_false
+      attribute.value = "butterfly"
+      attribute.changed?.should be_true
+    end
+
+    it "can detect change from and to a given value" do
+      attribute = Avram::Attribute.new(name: :color, param: nil, value: "red", param_key: "test_form")
+
+      attribute.value = "green"
+      attribute.changed?(from: "blue").should be_false
+      attribute.changed?(from: "red").should be_true
+      attribute.changed?(to: "red").should be_false
+      attribute.changed?(to: "green").should be_true
+      attribute.changed?(from: "green", to: "blue").should be_false
+      attribute.changed?(from: "red", to: "green").should be_true
+    end
```

Comment:
Could you add a spec where the original value is a strong and it is changed “to” nil? I think that might fail since the “to” argument will be nil

Reply:
You are right, this won't work. But while playing with this, I ran into a bigger issue. Attributes are not nillable. I quickly tested this with the following spec on the original master branch, which fails:

```cr
it "#value= is nillable" do
  color = Avram::Attribute.new(name: :color, param: nil, value: "mint", param_key: "test_form")
  color.value = nil
  color.value.should be_nil
end
```
Which results in:

```cr
In src/avram/attribute.cr:62:14

 62 | def value=(@value)
                 ^-----
Error: instance variable '@value' of Avram::Attribute(String) must be String, not Nil
```

I already have a solution ready. But before I go ahead and add the fix, am I right this should be nillable from string (or any other type for that matter)?

Reply:
@wout it is nila or if you explicitly set the type to nil. So if you do `Avram::Attribute(String?).new...` then it should let you set nil :)